indexing-service/src/main/java/org/apache/druid/indexing/overlord/ForkingTaskRunner.java

```diff
@@ -909,24 +909,9 @@ public CommandListBuilder add(String arg)
       return this;
     }
 
-    public CommandListBuilder addSystemProperty(String property, int value)
+    public CommandListBuilder addSystemProperty(String property, Object value)
```

Comment:
Using an `Object` as the value of a system property is not appropriate. It must always be a primitive or a String. Hence, the different `addSystemProperty()` methods.

As a workaround to the original problem, you can simply pass the integer value quoted as a String. (https://github.com/apache/druid/issues/16271#issuecomment-2058155166)

Reply:
```
```
How about calling the toString() method to make the public CommandListBuilder addSystemProperty (String property, String value) called